### PR TITLE
show concise image info for crictl ps

### LINKED ISSUE #360: display 13 characters for imageid  in "crictl ps"
We should display concise imageid in "crictl ps" command.  keeping consistent with "crictl images".
@Random-Liu @feiskyer 
```
$ crictl ps
CONTAINER ID        IMAGE                                                                     CREATED             STATE               NAME                      ATTEMPT
b7746a7c5d51b       sha256:6f7f2dc7fab5d7e7f99dc4ac176683a981a9ff911d643b9f29ffa146838deda3   29 minutes ago      Running             sidecar                   0
81065c78eb5a1       sha256:c2ce1ffb51ed60c54057f53b8756231f5b4b792ce04113c6755339a1beb25943   30 minutes ago      Running             dnsmasq                   0
ba71e307c802f       sha256:4689081edb103a9e8174bf23a255bfbe0b2d9ed82edc907abab6989d1c60f02c   31 minutes ago      Running             storage-provisioner       0

```

```
$ crictl images
IMAGE                                      TAG                 IMAGE ID            SIZE
gcr.io/k8s-minikube/storage-provisioner    v1.8.1              4689081edb103       20.7MB
k8s.gcr.io/etcd-amd64                      3.1.12              52920ad46f5bf       55.3MB
k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64     1.14.8              c2ce1ffb51ed6       11.1MB
k8s.gcr.io/k8s-dns-kube-dns-amd64          1.14.8              80cc5ea4b547a       13.4MB
k8s.gcr.io/k8s-dns-sidecar-amd64           1.14.8              6f7f2dc7fab5d       11.4MB
k8s.gcr.io/kube-addon-manager              v8.6                9c16409588eb1       21.2MB
k8s.gcr.io/kube-apiserver-amd64            v1.10.0             af20925d51a37       32.9MB
k8s.gcr.io/kube-controller-manager-amd64   v1.10.0             ad86dbed15559       30MB
k8s.gcr.io/kube-proxy-amd64                v1.10.0             bfc21aadc7d3e       31.5MB
k8s.gcr.io/kube-scheduler-amd64            v1.10.0             704ba848e69a7       12.9MB
k8s.gcr.io/kubernetes-dashboard-amd64      v1.8.1              e94d2f21bc0c2       40.5MB
k8s.gcr.io/pause                           3.1                 da86e6ba6ca19       317kB
```

### PR DESCRIPTION
**Pull request by yanxuean**
fix #360

### CODE REVIEW

File: cmd/crictl/container.go [hunk 1]
```diff
@@ -624,15 +624,11 @@ func ListContainers(client pb.RuntimeServiceClient, opts listOptions) error {
...
 		ctm := units.HumanDuration(time.Now().UTC().Sub(createdAt)) + " ago"
 		if !opts.verbose {
-			id := c.Id
-			if !opts.noTrunc {
-				id = strings.TrimPrefix(c.Id, "")
-				if len(id) > truncatedIDLen {
-					id = id[:truncatedIDLen]
-				}
-			}
+			id := getTruncatedID(opts.noTrunc, c.Id, "")
```
Comment (Random-Liu) [inline cmd/crictl/container.go]: I think this should be:
```
if !opts.noTrunc {
  id := getTruncatedID(c.Id, "")
}
```

It is weird that function `getTruncatedID` has a bool parameter to indicate returning a non-truncated id.

File: cmd/crictl/container.go [hunk 2]
```diff
@@ -624,15 +624,11 @@ func ListContainers(client pb.RuntimeServiceClient, opts listOptions) error {
...
+			// Now c.Image.Image is imageID in kubelet.
+			imageID := getTruncatedID(opts.noTrunc, c.Image.Image, "sha256:")
```
Comment (Random-Liu) [inline cmd/crictl/container.go]: Can we use `go-digest` to parse the id here? We should only truncate when it is an image id.

The kubelet implementation may change in the future, we should not rely on it.

File: cmd/crictl/container.go (hunk 1 above)
Comment (yanxuean) [reply cmd/crictl/container.go]: done

File: cmd/crictl/container.go (hunk 2 above)
Comment (yanxuean) [reply cmd/crictl/container.go]: done

**Comment by yanxuean** [comment]
```
[root@cloud-test2:/home/cloud]$ docker ps
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS               NAMES
9f7a48803a2f        9f76b554c615                 "/vms.sh"                21 hours ago        Up 21 hours                             k8s_vms_virtlet-wcb8r_kube-system_487bc52a-9c77-11e8-b87c-52540031cdce_0
b940b37a9f9e        9f76b554c615                 "/start.sh"              21 hours ago        Up 21 hours                             k8s_virtlet_virtlet-wcb8r_kube-system_487bc52a-9c77-11e8-b87c-52540031cdce_0
7fb15432579f        9f76b554c615                 "/libvirt.sh"            21 hours ago        Up 21 hours                             k8s_libvirt_virtlet-wcb8r_kube-system_487bc52a-9c77-11e8-b87c-52540031cdce_0
3f075e72dfa2        k8s.gcr.io/pause-amd64:3.1   "/pause"                 21 hours ago        Up 21 hours                             k8s_POD_virtlet-wcb8r_kube-system_487bc52a-9c77-11e8-b87c-52540031cdce_0
f60944900ceb        32609f1b11ae                 "/usr/local/bin/kube-"   21 hours ago        Up 21 hours                             k8s_kube-proxy_kube-proxy-bqnhm_kube-system_fad6d780-90a5-11e8-b3cf-52540031cdce_6
ab863e4ae6f3        k8s.gcr.io/pause-amd64:3.1   "/pause"                 21 hours ago        Up 21 hours                             k8s_POD_kube-proxy-bqnhm_kube-system_fad6d780-90a5-11e8-b3cf-52540031cdce_0
[root@cloud-test2:/home/cloud]$
[root@cloud-test2:/home/cloud]$ ./crictl ps -o table
CONTAINER ID        IMAGE               CREATED             STATE               NAME                ATTEMPT
9f7a48803a2f5       9f76b554c615d       22 hours ago        Running             vms                 0
b940b37a9f9ee       9f76b554c615d       22 hours ago        Running             virtlet             0
7fb15432579fa       9f76b554c615d       22 hours ago        Running             libvirt             0
f60944900ceb3       32609f1b11ae7       22 hours ago        Running             kube-proxy          6
[root@cloud-test2:/home/cloud]$ ./crictl pods
FATA[0000] listing pod sandboxes failed: unsupported output format ""
[root@cloud-test2:/home/cloud]$ ./crictl pods -o table
POD ID              CREATED             STATE               NAME                NAMESPACE           ATTEMPT
3f075e72dfa28       22 hours ago        Ready               virtlet-wcb8r       kube-system         0
ab863e4ae6f33       22 hours ago        Ready               kube-proxy-bqnhm    kube-system         0
[root@cloud-test2:/home/cloud]$ ./crictl images -o table
IMAGE                                                  TAG                 IMAGE ID            SIZE
busybox                                                latest              22c2dd5ee85dc       1.16MB
gcr.io/google_containers/busybox                       latest              e7d168d7db455       2.43MB
gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64   1.14.5              459944ce8cc4f       41.4MB
gcr.io/google_containers/k8s-dns-kube-dns-amd64        1.14.5              512cd7425a731       49.4MB
gcr.io/google_containers/k8s-dns-sidecar-amd64         1.14.5              fed89e8b4248a       41.8MB
gcr.io/google_containers/kube-proxy-amd64              v1.10.5             32609f1b11ae7       97.9MB
gcr.io/google_containers/pause-amd64                   3.0                 99e59f495ffaa       747kB
k8s.gcr.io/pause-amd64                                 3.1                 da86e6ba6ca19       742kB
mirantis/virtlet                                       latest              9f76b554c615d       891MB
[root@cloud-test2:/home/cloud]$

```

**Comment by yanxuean** [comment]
btw: Can we display podid in "crictl ps",  Now we don't know which pod these container belong to.

**Comment by Random-Liu** [comment]
> btw: Can we display podid in "crictl ps", Now we don't know which pod these container belong to.

SGTM

**Pull request merged**